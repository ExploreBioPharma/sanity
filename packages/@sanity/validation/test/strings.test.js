const {Rule} = require('../src')

describe('string', () => {
  test('returns no warnings/errors on valid string', () => {
    const result = Rule.string()
      .min(0)
      .max(5)
      .validate('hei')

    expect(result).toMatchSnapshot()
  })

  test('returns errors on min length constraint', () => {
    const result = Rule.string()
      .min(1)
      .validate('')

    expect(result).toMatchSnapshot()
  })

  test('returns errors on max length constraint', () => {
    const result = Rule.string()
      .max(5)
      .validate('heisann')

    expect(result).toMatchSnapshot()
  })
})
