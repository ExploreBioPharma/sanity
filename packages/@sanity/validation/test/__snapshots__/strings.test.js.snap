// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`string returns errors on max length constraint 1`] = `
Object {
  "errors": Array [
    [ValidationError: String must be at most 5 characters long],
  ],
  "warnings": Array [],
}
`;

exports[`string returns errors on min length constraint 1`] = `
Object {
  "errors": Array [
    [ValidationError: String must be at least 1 characters long,],
  ],
  "warnings": Array [],
}
`;

exports[`string returns no warnings/errors on valid string 1`] = `
Object {
  "errors": Array [],
  "warnings": Array [],
}
`;
